neo.UnitTests/SmartContract/Native/Tokens/UT_Nep5Token.cs

```diff
@@ -0,0 +1,77 @@
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Neo.Ledger;
+using Neo.Persistence;
+using Neo.SmartContract;
+using Neo.SmartContract.Native.Tokens;
+using Neo.UnitTests.Ledger;
+using Neo.VM;
+using System;
+
+namespace Neo.UnitTests.SmartContract.Native.Tokens
+{
+    [TestClass]
+    public class UT_Nep5Token
+    {
+        protected const byte Prefix_TotalSupply = 11;
+
+        [TestMethod]
+        public void TestTotalSupply()
+        {
+            var mockSnapshot = new Mock<Snapshot>();
+            var myDataCache = new TestDataCache<StorageKey, StorageItem>();
+            StorageItem item = new StorageItem
+            {
+                Value = new byte[] { 0x01 }
+            };
+            var key = CreateStorageKey(Prefix_TotalSupply);
+
+            var ServiceHash = "test".ToInteropMethodHash();
+            byte[] script = null;
+            using (ScriptBuilder sb = new ScriptBuilder())
+            {
+                sb.EmitSysCall(ServiceHash);
+                script = sb.ToArray();
+            }
+            var Hash = script.ToScriptHash();
+            key.ScriptHash = Hash;
+
+            myDataCache.Add(key, item);
+            mockSnapshot.SetupGet(p => p.Storages).Returns(myDataCache);
+            TestNep5Token test = new TestNep5Token();
+            ApplicationEngine ae = new ApplicationEngine(TriggerType.Application, null, mockSnapshot.Object, 0);
+            StackItem stackItem = test.TotalSupply(ae, null);
+            stackItem.GetBigInteger().Should().Be(1);
```

Comment:
Would be nice other test with `Decimals`

Reply:
It seems StackItem don't have a GetDecimals function and cannot convert the stackItem to decimal. So I don't know how to test with decimals.

Reply:
I means with the factor, at the end all is a big integer. 100_000_000 for gas, for example